### PR TITLE
Feature: add skeletons

### PR DESCRIPTION
**Pull request by fishtriangle**
Add BaseSkeleton and ProductListItemSkeleton components 
Add to tailwind.config.js keyframes for skeleton animation 
Add to CategoryView.tsx skeletons while categories are loading 
Add to ProductListItemSkeleton.tsx skeletons while products are loading

### CODE REVIEW

**Comment by Loskir** [comment]
😍

File: components/Skeletons/BaseSkeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        className={cx(
+          className,
+          "bg-gray-100 rounded-lg relative before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white before:to-transparent isolate overflow-hidden"
+        )}
+      />
+    </>
+  )
+}
+
+export default baseSkeleton
```
Comment (Loskir) [inline components/Skeletons/BaseSkeleton.tsx:17]: Maybe use named exports instead of default?

Comment (fishtriangle) [reply components/Skeletons/BaseSkeleton.tsx:17]: and index page in component folder?

Comment (Loskir) [reply components/Skeletons/BaseSkeleton.tsx:17]: no, like `export const BaseSkeleton` and then `import { BaseSkeleton }`

I believe it helps for autocomplete?

Comment (fishtriangle) [reply components/Skeletons/BaseSkeleton.tsx:17]: Ok, i'll change it

**Comment by Loskir** [comment]
Looks great! But you forgot to add them to CategoryView

**Comment by Loskir** [comment]
And also add skeletons to ProductView?

**Comment by fishtriangle** [comment]
Made some corrections!

**Comment by Loskir** [comment]
image

Skeletons for categories are misaligned

**Comment by Loskir** [comment]
Nevermind, can't reproduce locally

**Comment by Loskir** [comment]
Thank you so much! I'll merge this and add some corrections after that.

Also it looks like you're using prettier. Please install dprint formatter and use it for future contributions. The formatter itself is installed as a npm module (it is included in dev deps), you can also install a plugin for your IDE.

**Comment by fishtriangle** [comment]
> image
> 
> Skeletons for categories are misaligned

Test for that problem - and several times caught it. Looks like it doesnt recognize tailwind width utility in array that is not used anywhere else. Thats why some skeleton blocks have width = 0. I dont know why it works so, but temporary solution maybe add  `<div className={"w-6/12 w-5/12 w-4/12 hidden"} />` before array mapping

**Comment by Loskir** [comment]
Yeah it's tricky to get dynamic tailwind values to work because tailwind does tree shaking for unused classes and doesn't know which classes are used. You should probably rewrite it using your own classes or just css-in-js using emotion

**Comment by fishtriangle** [comment]
ok, i'll do with my own classes

**Comment by Loskir** [comment]
Great, please open another pr

**Pull request merged**